features/core/src/main/java/org/apache/karaf/features/internal/download/impl/DownloadManagerHelper.java

```diff
@@ -16,17 +16,37 @@
  */
 package org.apache.karaf.features.internal.download.impl;
 
+import java.util.Collection;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public final class DownloadManagerHelper {
 
-    private static final Pattern IGNORED_PROTOCOL_PATTERN = Pattern.compile("^(jar|war|war-i|warref|webbundle|wrap|spring|blueprint):.*$");
+    private static final String DEFAULT_IGNORED_PROTOCOL_PATTERN = "jar|war|war-i|warref|webbundle|wrap|spring|blueprint";
```

Comment:
What's the rational to not use the pattern ? It should be equivalent right ?

Reply:
Maybe it's worth to update the prefix pattern no ?

Reply:
From my perspective, this pattern is the "default pattern", it will be changed only if there exists a <extraProtocols> tag, which <params> values will be added to this pattern, by reconstructing the pattern on run time.

Reply:
It is allowing karaf-assembly-plugin users to define their own set of custom prefixes